### PR TITLE
Issue/376 patient bottom sheet

### LINKED ISSUE #376: Add PatientBottomSheet
### Describe the feature

Add PatientBottomSheet
Frame 156

### Why is this feature needed?

To see all tasks assigned to a patient

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by florian-sabonchi**
## Which issues does this pull request close?
closes #376

### CODE REVIEW

File: apps/tasks/lib/screens/main_screen_subscreens/my_tasks_screen.dart [hunk 1]
```diff
@@ -10,6 +10,7 @@ import 'package:tasks/dataclasses/subtask.dart';
 import 'package:tasks/screens/settings_screen.dart';
 import 'package:helpwave_widget/loading.dart';
 
+import '../../components/task_expansion_tile.dart';
```
Comment (DasProffi) [inline apps/tasks/lib/screens/main_screen_subscreens/my_tasks_screen.dart:13]: Remove the empty line
```suggestion
import 'package:helpwave_widget/loading.dart';
import '../../components/task_expansion_tile.dart';
```

File: apps/tasks/lib/components/task_card.dart [hunk 2]
```diff
@@ -62,86 +63,99 @@ class TaskCard extends StatelessWidget {
...
+
+    return GestureDetector(
+      onTap: () {
+        showModalBottomSheet(
+          context: context,
+          isScrollControlled: true,
+          builder: (BuildContext context) => SizedBox(
+            height: width * 0.8,
+            child: PatientBottomSheet(task: task),)
+        );
```
Comment (DasProffi) [inline apps/tasks/lib/components/task_card.dart:76]: This should not open the patient screen

File: apps/tasks/lib/components/patient_bottom_sheet.dart [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+import 'package:helpwave_theme/constants.dart';
+import 'package:helpwave_theme/theme.dart';
+import 'package:provider/provider.dart';
+import 'package:tasks/components/task_expansion_tile.dart';
+import 'package:tasks/dataclasses/task.dart';
+
+class PatientBottomSheet extends StatefulWidget {
+  const PatientBottomSheet({Key? key, required this.task}) : super(key: key);
+
+  final TaskWithPatient task;
```
Comment (DasProffi) [inline apps/tasks/lib/components/patient_bottom_sheet.dart:12]: The Tasks for the patient will be loaded by the patient id, only require the patient

File: apps/tasks/lib/components/patient_bottom_sheet.dart [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+                            title: Text(
+                              overflow: TextOverflow.ellipsis,
+                              "Room 2 - Bed 1", // TODO: replace with real data
+                              style: TextStyle(color: Colors.grey),
+                            ),
+                            children: [
+                            ],
+                          ),
+                        ),
+                      )
```
Comment (DasProffi) [inline apps/tasks/lib/components/patient_bottom_sheet.dart:75]: Use a dropdown and decrease the padding to the title

File: apps/tasks/lib/components/patient_bottom_sheet.dart [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+                    ],
+                  ),
+                ),
+                TextFormField(
```
Comment (DasProffi) [inline apps/tasks/lib/components/patient_bottom_sheet.dart:79]: There should be a Notes title here

File: apps/tasks/lib/components/patient_bottom_sheet.dart [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+                  ),
+                  maxLines: 6,
+                  keyboardType: TextInputType.multiline,
+                ),
+                Padding(
+                  padding: const EdgeInsets.symmetric(vertical: paddingMedium, horizontal: paddingOffset ),
+                  child:  Row(
+                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                    children: [
+                      Text(context.localization!.tasks, style: const TextStyle(fontSize: fontSizeBig, fontWeight: FontWeight.bold),),
```
Comment (DasProffi) [inline apps/tasks/lib/components/patient_bottom_sheet.dart:92]: Tasks should be capitalized here

File: apps/tasks/lib/components/patient_bottom_sheet.dart [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+                TextFormField(
+                  initialValue: widget.task.notes,
+                  decoration: InputDecoration(
+                    hintText: context.localization!.notes,
+                  ),
+                  maxLines: 6,
+                  keyboardType: TextInputType.multiline,
+                ),
+                Padding(
+                  padding: const EdgeInsets.symmetric(vertical: paddingMedium, horizontal: paddingOffset ),
```
Comment (DasProffi) [inline apps/tasks/lib/components/patient_bottom_sheet.dart:88]:
```suggestion
                  padding: const EdgeInsets.only(top: paddingMedium, left: paddingOffset, right: paddingOffset, bottom: paddingSmall),
```

File: apps/tasks/lib/components/patient_bottom_sheet.dart [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+                          color: primaryColor
+                        ),
+                        child: IconButton(
+                          onPressed: () {},
+                          icon: const Icon(size: 25 , Icons.add, color: Colors.white,),
+                        ),
+                      )
+                    ],
+                  ),
+                ),
```
Comment (DasProffi) [inline apps/tasks/lib/components/patient_bottom_sheet.dart:107]: Make the icon smaller maybe iconSizeSmall for the outer ring and iconSizeTiny for the add

**Review by DasProffi — CHANGES_REQUESTED** [review]

**Comment by DasProffi** [comment]
The padding isn't aligning horizontally, please use the same padding

**Comment by DasProffi** [comment]
This issue is also very large, if you want to move some part like the grpc requests to a new issue, please open one and link it here

File: apps/tasks/lib/components/patient_bottom_sheet.dart (hunk 6 above)
Comment (florian-sabonchi) [reply apps/tasks/lib/components/patient_bottom_sheet.dart:100]: According to the mockups not

Comment (DasProffi) [reply apps/tasks/lib/components/patient_bottom_sheet.dart:103]: Frame 156 (1)

Comment (florian-sabonchi) [reply apps/tasks/lib/components/patient_bottom_sheet.dart:103]: Sorry, I think was a misunderstanding

**Comment by DasProffi** [comment]
- there is an overflow at the bottom
- the top row icons arent't aligned with the notes text box
- tasks is still in lower case
- the tasks List view has dividers which shouldn't be there
image

**Comment by florian-sabonchi** [comment]
> * s still in lower case

You need run melos run intl again to apply new spelling

**Comment by florian-sabonchi** [comment]
can't reproduce overflow and tailing issue

**Comment by florian-sabonchi** [comment]
https://github.com/helpwave/mobile-app/assets/54689374/fd08931b-d815-4a8d-8503-4166c78c1515

**Comment by DasProffi** [comment]
> Screen_recording_20231008_165601.mp4

There is still the alignment which isn't matching and the dividers in the ListView

**Comment by MaxSchaefer** [comment]
Is this component already in use? I cannot trigger the opening in the app. It seems like that there is no way to test this right now without going manually into the code and placing this component.

**Comment by florian-sabonchi** [comment]
> Is this component already in use? I cannot trigger the opening in the app. It seems like that there is no way to test this right now without going manually into the code and placing this component.

No currently not

**Comment by florian-sabonchi** [comment]
> Is this component already in use? I cannot trigger the opening in the app. It seems like that there is no way to test this right now without going manually into the code and placing this component.

No currently not

**Comment by DasProffi** [comment]
Newly opened issues which will not be addressed within this PR:
- #392 
- #393 
- #394

**Review by DasProffi — APPROVED** [review]

**Pull request merged**